### ISSUE TITLE
split args.error into args.internal_error and original

### ISSUE THREAD

**Issue by brab** [mozilla/metrics-graphics#543]
The "No data was supplied" error is correctly set when a graph is loaded with no data, but not cleared when the same graph is later loaded with data.

**Comment by hamilton** [comment]
Thanks for the issue. Are you using the 2.7 release or the latest commit?

**Comment by brab** [comment]
I'm using the 2.7 release; I didn't check master to see if it was already fixed.

**Comment by hamilton** [comment]
probably isn't yet unless someone else patched it :)

**Comment by hamilton** [comment]
@almossawi this bug demonstrates a potential design issue with how we use ``args.error``. Right now we are conflating user-defined errors (for instance, a user may display the exclam on the chart title if there is a context-specific error the user wishes to communicate to the consumer) vs. graph-defined errors (such as not passing in data). 

I noticed this because while trying to clear args.error in line.js (~ line 10) I triggered a failed test in ``chart_title_test.js``, which sets a custom error.

I feel that we should maybe decouple these two use cases. We wouldn't need to change the API, but would need to split args.error into something like ``args.error`` (user-defined in API) and ``args.internal_error`` (which the user does not touch) or something like that. Thoughts?
